test/refactoring/ConvertAsync/convert_pattern.swift

```diff
@@ -5,6 +5,412 @@ enum E : Error { case e }
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=INLINE-CASE-TYPED %s
+  stringTupleParam { strs, err in
+    guard case let (str1, str2)?: (String, String)? = strs else { return }
+    print(str1, str2)
+  }
+
+  // INLINE-CASE-TYPED:      let (str1, str2) = try await stringTupleParam()
+  // INLINE-CASE-TYPED-NEXT: print(str1, str2)
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=OUT-OF-LINE %s
+  stringTupleParam { strs, err in
+    guard let (str1, str2) = strs else { return }
+    print(str1, str2, strs!)
+  }
+
+  // OUT-OF-LINE:      let strs = try await stringTupleParam()
+  // OUT-OF-LINE-NEXT: let (str1, str2) = strs
+  // OUT-OF-LINE-NEXT: print(str1, str2, strs)
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=OUT-OF-LINE-VAR %s
+  stringTupleParam { strs, err in
+    guard var (str1, _) = strs else { return }
+    str1 = ""
+    print(str1, {strs!})
+  }
+
+  // OUT-OF-LINE-VAR:      let strs = try await stringTupleParam()
+  // OUT-OF-LINE-VAR-NEXT: var (str1, _) = strs
+  // OUT-OF-LINE-VAR-NEXT: str1 = ""
+  // OUT-OF-LINE-VAR-NEXT: print(str1, {strs})
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=OUT-OF-LINE-CASE %s
+  stringTupleParam { strs, err in
+    guard case (let str1, var str2)? = strs else { return }
+    print(str1, str2, strs!)
+  }
+
+  // OUT-OF-LINE-CASE:      let strs = try await stringTupleParam()
+  // OUT-OF-LINE-CASE-NEXT: var (str1, str2) = strs
+  // OUT-OF-LINE-CASE-NEXT: print(str1, str2, strs)
+
+  // RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=FALLBACK %s
+  stringTupleParam { strs, err in
+    guard let (str1, str2) = strs else { return }
+    print(str1, str2)
+    if .random(), err == nil {
+      print("yay")
+    } else {
+      print("nay")
+    }
+  }
+
+  // FALLBACK:      var strs: (String, String)? = nil
+  // FALLBACK-NEXT: var err: Error? = nil
+  // FALLBACK-NEXT: do {
+  // FALLBACK-NEXT:   strs = try await stringTupleParam()
+  // FALLBACK-NEXT: } catch {
+  // FALLBACK-NEXT:   err = error
+  // FALLBACK-NEXT: }
+  // FALLBACK-EMPTY:
+  // FALLBACK-NEXT: guard let (str1, str2) = strs else { <#return#> }
+  // FALLBACK-NEXT: print(str1, str2)
+  // FALLBACK-NEXT: if .random(), err == nil {
+  // FALLBACK-NEXT:   print("yay")
+  // FALLBACK-NEXT: } else {
+  // FALLBACK-NEXT:   print("nay")
+  // FALLBACK-NEXT: }
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=MIXED-BINDINGS %s
+  stringTupleParam { strs, err in
```

Comment:
I wonder how often this actually comes up 😆. Very nice though!

Reply:
Yeah this exact case probably won't come up haha, but the subset of cases it covers might do, I just wanted to make sure they all interacted with each other okay :)